### PR TITLE
Fix shuffle questions

### PR DESCRIPTION
**Pull request by adityacp**
This PR fixes shuffled questions bug. 
Now if shuffled questions option is enabled, Answer paper includes shuffled questions irrespective of Question paper questions order.

### CODE REVIEW

File: yaksh/models.py [hunk 1]
```diff
@@ -851,6 +852,8 @@ def _get_questions_for_answerpaper(self):
         questions = self.get_ordered_questions()
         for question_set in self.random_questions.all():
             questions += question_set.get_random_questions()
+        if self.shuffle_questions:
+            return self.get_shuffled_questions(questions)
```
Comment (prabhuramachandran) [inline yaksh/models.py]: Better to have the return in an else

File: yaksh/models.py [hunk 2]
```diff
@@ -941,9 +947,19 @@ def get_ordered_questions(self):
...
 
+    def get_shuffled_questions(self, questions):
+        """Get shuffled questions if auto suffle is enabled"""
+        random.shuffle(questions)
+        return questions
+
+    def has_questions(self):
+        questions = self.get_ordered_questions() + \
+                    list(self.random_questions.all())
+        return False if len(questions) == 0 else True
```
Comment (prabhuramachandran) [inline yaksh/models.py]: Just `return len(questions) > 0`

File: yaksh/models.py [hunk 3]
```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
         default='inprogress'
     )
 
+    # set question order
+    questions_order = models.CharField(max_length=255, blank=True, default='')
```
Comment (prabhuramachandran) [inline yaksh/models.py]: This is too short a length, what if you have a paper with 100 questions?

File: yaksh/models.py [hunk 4]
```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
...
+
     objects = AnswerPaperManager()
 
     def current_question(self):
         """Returns the current active question to display."""
         if self.questions_unanswered.all():
-            return self.questions_unanswered.all()[0]
-        return self.questions.all()[0]
+            cur_question = self.get_current_question(
+                self.questions_unanswered.all())
```
Comment (prabhuramachandran) [inline yaksh/models.py]: You are making two queries here so save the unanswered_questions here and then use that.

File: yaksh/models.py [hunk 5]
```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
...
+        else:
+            cur_question = self.get_current_question(self.questions.all())
+        return cur_question
+
+    def get_current_question(self, questions):
+        if self.questions_order:
+            question_id = int(self.questions_order.split(',')[0])
```
Comment (prabhuramachandran) [inline yaksh/models.py]: This logic is not clear.  Why is the first question in the questions_order the current one?

File: yaksh/models.py [hunk 6]
```diff
@@ -1164,13 +1180,27 @@ class AnswerPaper(models.Model):
...
-            return self.questions_unanswered.all()[0]
-        return self.questions.all()[0]
+            cur_question = self.get_current_question(
+                self.questions_unanswered.all())
+        else:
+            cur_question = self.get_current_question(self.questions.all())
+        return cur_question
+
+    def get_current_question(self, questions):
+        if self.questions_order:
```
Comment (prabhuramachandran) [inline yaksh/models.py]: Why is this if case needed at all, don't you always have a questions_order when you make the paper or is there a case when this doesn't happen?  Maybe there is I am just asking.

File: yaksh/models.py (hunk 5 above)
Comment (prabhuramachandran) [reply yaksh/models.py]: Isn't there some other database entry that tells you which is the active question?

File: yaksh/models.py (hunk 6 above)
Comment (adityacp) [reply yaksh/models.py]: If this condition is not added then older answer papers where there are no question order will not be able to get questions.

File: yaksh/models.py (hunk 5 above)
Comment (adityacp) [reply yaksh/models.py]: No we don't store the current question. we just take the first question from the list of questions.
Then for next question we just iterate over that list based on current question id.

File: yaksh/models.py (hunk 3 above)
Comment (adityacp) [reply yaksh/models.py]: I can make it a TextField.

**Pull request merged**